python/cuml/test/dask/test_datasets.py

```diff
@@ -76,3 +76,55 @@ def test_make_blobs(nrows,
 
     finally:
         c.close()
+
+
+@pytest.mark.parametrize('n_samples', [100, 100000])
+@pytest.mark.parametrize('n_features', [10, 100])
+@pytest.mark.parametrize('n_informative', [7])
+@pytest.mark.parametrize('n_targets', [1, 3])
+@pytest.mark.parametrize('bias', [-4.0])
+@pytest.mark.parametrize('effective_rank', [None, 6])
+@pytest.mark.parametrize('tail_strength', [0.5])
+@pytest.mark.parametrize('noise', [3.5])
+@pytest.mark.parametrize('shuffle', [True, False])
+@pytest.mark.parametrize('coef', [True, False])
+@pytest.mark.parametrize('random_state', [None, 1234])
+@pytest.mark.parametrize('n_parts', [1, 3])
+def test_make_regression(n_samples, n_features, n_informative,
+                         n_targets, bias, effective_rank,
+                         tail_strength, noise, shuffle,
+                         coef, random_state, n_parts,
+                         cluster):
+    c = Client(cluster)
+    try:
+        from cuml.dask.datasets import make_regression
+
+        result = make_regression(n_samples=n_samples, n_features=n_features,
+                                 n_informative=n_informative,
+                                 n_targets=n_targets, bias=bias,
+                                 effective_rank=effective_rank, noise=noise,
+                                 shuffle=shuffle, coef=coef,
+                                 random_state=random_state, n_parts=n_parts)
+
+        if coef:
+            out, values, coefs = result
+        else:
+            out, values = result
+
+        assert out.shape == (n_samples, n_features), "out shape mismatch"
+
+        if n_targets > 1:
+            assert values.shape == (n_samples, n_targets), \
+                   "values shape mismatch"
+        else:
+            assert values.shape == (n_samples,), "values shape mismatch"
+
+        if coef:
+            if n_targets > 1:
+                assert coefs.shape == (n_features, n_targets), \
+                       "coefs shape mismatch"
+            else:
+                assert coefs.shape == (n_features,), "coefs shape mismatch"
+
+    finally:
+        c.close()
```

Comment:
An additional small test or checks like https://github.com/scikit-learn/scikit-learn/blob/ac9c3e1d33ebd771d4598eeb6cda61ce9d9a70de/sklearn/datasets/tests/test_samples_generator.py#L252 and https://github.com/scikit-learn/scikit-learn/blob/ac9c3e1d33ebd771d4598eeb6cda61ce9d9a70de/sklearn/datasets/tests/test_samples_generator.py#L255 might be benefitial if possible

Reply:
@dantegd beat me to this- I was going to recommend using some of the tests scikit-learn uses as well. It's a good verification that we match their API. Validating shapes isn't going to be enough- you'll also want to have good tests that make sure the regression data actually behaves according to the remaining parameters.